chcemvediet/apps/anonymization/models.py

```diff
@@ -100,14 +99,105 @@ def save(self, *args, **kwargs):
+            help_text=squeeze(u"""
+                Empty filename if recognition failed or recognition didn't create any file.
+                """))
+
+    # May be empty: Extension automatically adjusted in save() when creating new object.
+    name = models.CharField(max_length=255, blank=True,
+            help_text=squeeze(u"""
+                Attachment file name, e.g. "document.pdf". Extension automatically adjusted when
+                creating a new object. Empty, if file.name is empty.
+                """))
+
+    # May be NULL
+    content_type = models.CharField(max_length=255, null=True,
+            help_text=squeeze(u"""
+                Attachment content type, e.g. "application/pdf". The value may be specified even if
+                recognition failed.
+                """))
+
+    # May NOT be NULL; Automatically computed in save() when creating a new object if undefined.
+    created = models.DateTimeField(blank=True,
+            help_text=squeeze(u"""
+                Date and time the attachment was recognized. Leave blank for current time.
+                """))
+
+    # May be NULL; Automatically computed in save() when creating a new object.
+    size = models.IntegerField(null=True, blank=True,
+            help_text=squeeze(u"""
+                Attachment file size in bytes. NULL if file is NULL. Automatically computed when
+                creating a new object.
+                """))
+
+    # May NOT be NULL
+    debug = models.TextField(blank=True,
+            help_text=squeeze(u"""
+                Debug message from recognition.
+                """))
+
+    # Backward relations added to other models:
+    #
+    #  -- Attachment.attachment_normalization_set
+    #     May be empty
+
+    # Indexes:
+    #  -- attachment: ForeignKey
+
+    objects = AttachmentRecognitionQuerySet.as_manager()
+
+    @cached_property
+    def content(self):
+        if not self.file:
+            return None
+        try:
+            self.file.open(u'rb')
+            return self.file.read()
+        except IOError:
+            logger = logging.getLogger(u'chcemvediet.apps.anonymization')
+            logger.error(u'{} is missing its file: "{}".'.format(self, self.file.name))
+            raise
+        finally:
+            self.file.close()
+
+    @decorate(prevent_bulk_create=True)
+    def save(self, *args, **kwargs):
+        if self.pk is None:  # Creating a new object
+            if self.created is None:
+                self.created = utc_now()
+            if self.file._file:
+                self.file.name = random_string(10)
+                self.size = self.file.size
+                self.name = adjust_extension(self.attachment.name, self.content_type)
+        super(AttachmentRecognition, self).save(*args, **kwargs)
+
+    def __unicode__(self):
+        return format(self.pk)
+
 @datacheck.register
-def datachecks(superficial, autofix):
+def datachecks_AttachmentNormalization(superficial, autofix):
```

Comment:
Je lepsie mat dve funkcie s @datacheck.register alebo to dam do jednej ? 
Prepisem nazov funkcie analogicky aj v subore `attachments/models.py`

Reply:
Imho takto oddelene su citatelnejsie.